src/components/CompanyItem/CompanyItem.tsx

```diff
@@ -0,0 +1,63 @@
+import { flushSync } from 'react-dom';
+import { useNavigate } from 'react-router-dom';
+
+import placeholder from '../../assets/placeholder.svg';
+import { CompanyData } from '../../types/companyData';
+
+type CompanyItemProps = {
+	companyItem: CompanyData;
+	navigateTo: string;
+	grayscale?: boolean;
+};
+
+export default function CompanyItem({
+	companyItem,
+	navigateTo,
+	grayscale = false,
+}: CompanyItemProps) {
+	const navigate = useNavigate();
+
+	const handleClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
```

Comment:
I prefer to import { MouseEvent } from React and use only

 ```suggestion
	const handleClick = (event: MouseEvent<HTMLAnchorElement>) => {
```
But both works.